presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorPageSinkProvider.java

```diff
@@ -19,9 +19,11 @@
 import com.facebook.presto.spi.ConnectorSession;
 import com.facebook.presto.spi.PageSinkProperties;
 
+import java.util.Optional;
+
 public interface ConnectorPageSinkProvider
 {
-    ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorOutputTableHandle outputTableHandle, PageSinkProperties pageSinkProperties);
+    ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorOutputTableHandle outputTableHandle, Optional<ConnectorDistributedMetadataUpdater> distributedMetadataUpdater, PageSinkProperties pageSinkProperties);
 
-    ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorInsertTableHandle insertTableHandle, PageSinkProperties pageSinkProperties);
+    ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorInsertTableHandle insertTableHandle, Optional<ConnectorDistributedMetadataUpdater> distributedMetadataUpdater, PageSinkProperties pageSinkProperties);
```

Comment:
I think we can directly inject `ConnectorDistributedMetadataUpdater` into the underlying page sink provider then passing through SPI? The design principle is to keep spi as simple as possible

Reply:
Got it.